tests/ServerTest.php

```diff
@@ -1254,6 +1254,40 @@ function ($data) use (&$buffer) {
         $this->assertContains("\r\n\r\nError 431: Request Header Fields Too Large", $buffer);
     }
 
+    public function testCustomRequestHeaderSizeIsPassedAndNoHeaderOverflowErrorWillAppear()
+    {
+      $maxSize = 1024 * 16;
+
+      $options = array();
+      $options['max_header_size'] = $maxSize;
+
+      $server = new Server(function (ServerRequestInterface $request) {
+        return new Response(200, array());
+      }, $options);
+
+      $buffer = '';
+
+      $this->connection
+        ->expects($this->any())
+        ->method('write')
+        ->will(
+          $this->returnCallback(
+            function ($data) use (&$buffer) {
+              $buffer .= $data;
+            }
+          )
+        );
+
+      $server->listen($this->socket);
+      $this->socket->emit('connection', array($this->connection));
+
+      $data = "GET / HTTP/1.1\r\nHost: example.com\r\nConnection: close\r\nX-DATA: ";
+      $data .= str_repeat('A', 4097 - strlen($data)) . "\r\n\r\n";
```

Comment:
Why 4097 here? I'd expect `$maxSize - strlen($data)` there.

Reply:
I was expecting that the default value is not throwing the overflow anymore. Your example makes more sense. I will adopt and check the other overflow tests to simply rely on those 3 test-cases:

* check the default overflow `$defaultValue + 1`
* check a custom header-size-value fits `$maxSize - strlen($data)`
* check a custom header-size will overflow `$maxSize + 1 - strlen($data)`